packages/prisma-client/migrations-cli/umzug.ts

```diff
@@ -0,0 +1,203 @@
+  // But this is not entirelly safe as it will fail if a query contains ; as a part of a string.
+  const queries = source
+    .split(";")
+    .map((q) => q.trim())
+    .filter((q) => q.length > 0);
+
+  await prisma.$transaction(
+    async (prisma) => {
+      for (const query of queries) {
+        await prisma.$executeRawUnsafe(query);
+      }
+    },
+    { timeout }
+  );
+};
+
+const tmpMemoryStorage = memoryStorage();
+
+export const migrator = new Umzug({
+  migrations: {
+    glob: ["../prisma/migrations/**/*.{ts,sql}", { cwd: __dirname }],
+    resolve(params) {
+      const filePath = params.path;
+
+      // for TypeScript
+      if (filePath == null) {
+        throw new Error("Path is required");
+      }
+
+      const name = path.dirname(filePath).split(path.sep).pop() as string;
+
+      if (path.extname(filePath) === ".sql") {
+        return {
+          name: name,
+          path: filePath,
+          up: async () => {
+            // throw new Error("TEST");
+            // await executeFile(filePath);
+          },
+        };
+      }
+
+      return { ...Umzug.defaultResolver(params), name };
+    },
+  },
+  context: {},
+  storage: {
+    logMigration(params) {
+      // @todo: update in _prisma_migrations as finished
+      return tmpMemoryStorage.logMigration(params);
+    },
+    unlogMigration() {
+      // this is for rollbacks, that we don't currently support
+      throw new Error("unlogMigration not implemented");
+    },
+    executed(meta) {
+      // @todo: read from _prisma_migrations
+      return tmpMemoryStorage.executed(meta);
+    },
+  },
+  create: {
+    template: (filepath) => {
+      const date = new Date();
+      const prefix =
+        date.getFullYear() +
+        [
+          date.getMonth() + 1,
+          date.getDate(),
+          date.getHours(),
+          date.getMinutes(),
+          date.getSeconds(),
+        ]
+          .map((item) => item.toString().padStart(2, "0"))
+          .join("");
+      const name = path.basename(filepath);
+      return [
+        [
+          path.join(
+            __dirname,
+            "../prisma/migrations",
+            `${prefix}_${name}`,
+            "migration.ts"
+          ),
+          fs.readFileSync(path.join(__dirname, "template.txt")).toString(),
+        ],
+      ];
+    },
+  },
+  logger: {
+    info: (event) => {
+      if (event.event === "migrating") {
+        console.log(`starting ${event.name}`);
+        return;
+      }
+      if (event.event === "migrated") {
+        console.log(`${event.name} done in ${event.durationSeconds}s`);
+        return;
+      }
+      if (event.event === "up") {
+        console.log(event.message);
+        return;
+      }
+      if (event.event === "created") {
+        console.log(`created: ${event.path}`);
+        return;
+      }
+      console.log(event);
+    },
+    error: (event) => console.error(event),
```

Comment:
probably could use process.stdout/stderr so we can keep console stuff for debugging, we have a linter that prevents us checking in console calls that should have warned about these

Reply:
I've now added a logger that uses console. The idea is that for debug you use console as normal, and for CLI output you use the logger.